icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -413,7 +418,8 @@ class Parser {
             }
             switch (token.getMatch()) {
                 case COMPOUND_PART_PER:
-                    if (fAfterPer) {
+                    if (fSawAnd) {
+                        // Mixed units can only contain single units.
```

Comment:
This is according to the current proposed LDML spec.

We need to keep an eye on https://unicode-org.atlassian.net/browse/CLDR-13700 though, since that might make things quite a bit more complicated. For this, we probably need an ICU ticket as counterpart, which we can refer to in a TODO, is that correct?